sns/parity_math.c

```diff
@@ -499,30 +953,87 @@ static void reed_solomon_encode(struct m0_parity_math *math,
 	}
 
 #undef PARITY_MATH_REGION_ENABLE
+
+	return M0_RC(0);
+}
+#endif /* !ISAL_ENCODE_ENABLED */
+
+#if ISAL_ENCODE_ENABLED
```

Comment:
The style with a large number of #if-s is difficult to read. Please move all isal code in a new .c file. Something like:

```
#if USE_ISAL
void isal_encode(...)
{
        ...
}
...
#else 
void isal_encode(...)
{
        M0_IMPOSSIBLE("ISAL is not configured.");
}
...
#endif 
```

Reply:
Code updated. Now many functions have two separate implementations for using Galois and Intel ISA library. At a time any one of the function is used depending on whether Intel ISA library is present or not. If Intel ISA library is present, it will use implementation specific to Intel ISA and use Intel ISA APIs for encoding and recovery, else it will use Galois library arithmetic functions.

e.g.

```
#if ISAL_ENCODE_ENABLED
static void reed_solomon_encode(struct m0_parity_math *math,
				const struct m0_buf *data,
				struct m0_buf *parity)
{
        /* Intel ISA specific implementation */
}
#else 
static void reed_solomon_encode(struct m0_parity_math *math,
				const struct m0_buf *data,
				struct m0_buf *parity)
{
        /* Reed Solomon algorithm uses Galois arithmetic APIs */
}
#endif /* #if ISAL_ENCODE_ENABLED */

```